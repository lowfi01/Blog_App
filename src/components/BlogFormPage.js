import React from 'react';
import PropTypes from 'prop-types';

class BlogFormPage extends React.Component{
  constructor(props){
    super(props);

    this.state = {
      title: props.blog ? props.blog.title : '',
      content: props.blog ? props.blog.content : '',
      date: props.date ? props.blog.date : ''
    }
  }

  onTitleChange = (e) => {
    const title = e.target.value;
    this.setState(() => ({
      title
    }))
  };

  onContentChange = (e) => {
    const content = e.target.value;
    this.setState(() => ({
      content
    }))
  };

  onSubmit = (e) => {
    e.preventDefault();
      this.props.onSubmit({
        ...this.state,
        date: Date.now()
      });
  }

  render() {
    return (
      <div>
        <form
          className="form"
          onSubmit={this.onSubmit}
        >
        <input
          className="text-input"
          type="text"
          value={this.state.title}
          onChange={this.onTitleChange}
          placeholder="Catchy Title.."
        />
        <textarea
          type="text"
          className="text-area"
          value={this.state.content}
          onChange={this.onContentChange}
          placeholder="...."
        />

        <button
          className="button"
          type="submit"
        > Save & Publish </button>

      </form>
    </div>
    )
  }
}


BlogFormPage.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  title: PropTypes.string,
  content: PropTypes.string
}

export default BlogFormPage;