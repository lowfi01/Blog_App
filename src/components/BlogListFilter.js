import React from 'react';
import {connect} from 'react-redux'
import {setTextFilter} from '../actions/filters';



class BlogListFilter extends React.Component {
  render(){
    return (
      <div className="content-container">
        <div className="input-group">

          <input
            type="text"
            className="text-input"
            placeholder="Search Blog Titles.."
            value={this.props.filters.text}
            onChange={e => {
              this.props.dispatch(setTextFilter(e.target.value))}
              }
            />

        </div>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  }
}

export default connect(mapStateToProps)(BlogListFilter);
