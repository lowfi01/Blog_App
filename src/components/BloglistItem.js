import React from 'react';
import PropTypes from 'prop-types';
import Link from 'react-router-dom/Link';
import {distanceInWords} from 'date-fns';


const BloglistItem = ({id, title, content, remove, date}) => (
    <Link
      className="list-item"
      to={`/edit/${id}`}
    >
      <div>
          <h3 className="list-item__title">{title} {distanceInWords(new Date(date))}</h3>
          <span className="missing">
          {content}
          </span>
          <button onClick={(e) => {
            e.preventDefault();
            remove(id);
          }
          }>
            Remove
          </button>
      </div>
    </Link>
  )


BloglistItem.propTypes = {
  title: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  remove: PropTypes.func.isRequired,
  date: PropTypes.string.isRequired
}

BloglistItem.defaultProps = {
  title: '',
  content: ''
}

export default BloglistItem;