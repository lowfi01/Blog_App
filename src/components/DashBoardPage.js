import React from 'react';

export const DashBoardPage = (props) => (
    <div>
      Dashboard Page Content
    </div>

  )

export default DashBoardPage;