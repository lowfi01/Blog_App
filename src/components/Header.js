import React from 'react';
import {connect} from 'react-redux';
import { Link } from 'react-router-dom';
import { startLogout } from '../actions/auth';

export const Header = (props) => (

  <div className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" to="/" activeClassName="is-active">
          <h1>Blog App</h1>
        </Link>
        <Link to="/create"><h2>Create</h2></Link>
        <button className="button button--link" onClick={ () => {
          // props.startLogout();
          }}>Logout</button>
      </div>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
