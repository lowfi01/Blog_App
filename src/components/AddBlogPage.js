import React from 'react';
import {connect} from 'react-redux';

import BlogFormPage from './BlogFormPage';
import {startAddBlogPost} from '../actions/blog';

const AddBlogPage = (props) => {
  const onSubmit = (blog) => {
    props.dispatch(startAddBlogPost(blog));
    props.history.push('/');
  }

  return (
    <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Create Post</h1>
          </div>
        </div>

        <div className="content-container">
          <BlogFormPage onSubmit={onSubmit} />
        </div>

      </div>
  )
}

export default connect()(AddBlogPage);