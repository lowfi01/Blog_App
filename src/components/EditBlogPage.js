import React from 'react';
import {connect} from 'react-redux';

import BlogFormPage from './BlogFormPage';
import {startEditBlogPost} from '../actions/blog';

const EditBlogPage = (props) => {
  const onSubmit = (state) => {
    props.dispatch(startEditBlogPost(props.match.params.id, state ))
    props.history.push('/');
  }

  return (
    <div>
      <h1>Edit your Blog</h1>
      <BlogFormPage
        blog={props.blog}
        onSubmit={onSubmit}
        />
    </div>
  )
}

const mapStateToProps = (state, props) => ({
    blog: state.blogs.find((blog) => {
     return blog.id === props.match.params.id
    })
  })


export default connect(mapStateToProps)(EditBlogPage);