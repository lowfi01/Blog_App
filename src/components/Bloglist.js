import React from 'react';
import {connect} from 'react-redux';

import BloglistItem from './BloglistItem';
import { startRemoveBlogPost } from '../actions/blog';
import filters from '../selectors/filter';

const Bloglist = (props) => {
    const remove = (id) => {
      props.dispatch(startRemoveBlogPost(id))
    }
    return (
      <div className="content-container">
        <div >
        {
          props.blogs.length === 0 ? (
            <p>No Blogs Created</p>
          ) : (
            props.blogs.map((data) => <BloglistItem key={data.id} {...data} remove={remove} />)
          )
        }
        </div>
      </div>
    )


  }

const mapStateToProps = (state) => ({
  blogs: filters(state.blogs, state.filters)
})

export default connect(mapStateToProps)(Bloglist);