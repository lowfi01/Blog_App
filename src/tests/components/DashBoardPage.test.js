import React from 'react';
import {shallow} from 'enzyme';
import { DashBoardPage } from '../../components/DashBoardPage';

test('should render Dash Board page', () => {
  const wrapper = shallow(<DashBoardPage/>);
  expect(wrapper).toMatchSnapshot();
});