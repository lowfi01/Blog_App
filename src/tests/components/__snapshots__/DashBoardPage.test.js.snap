// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Dash Board page 1`] = `
<div>
  Dashboard Page Content // add your components here :D
</div>
`;
